api/services/registration_service.py

```diff
@@ -52,7 +52,13 @@ def handle_registration(self, jsonData):
                         )
                     )
                     if user_program:
-                        models.UserProgram.query.update(user_program, user_program_data)
+                        if (
+                            user_program_data["program_id"]
+                            != app.config["DEFAULT_PROGRAM_ID"]
+                        ):
+                            models.UserProgram.query.update(
+                                user_program, user_program_data
+                            )
```

Comment:
Can we just write this condition?  I think this should be better

```
if user_program:
    if has_default_program_selection and user_program.program_id:
        user_program_data["program_id"] = user_program.program_id

Reply:
Yes, But we can also write like this.
```
if user_program:
    if has_default_program_selection:
        user_program_data["program_id"] = user_program.program_id
    models.UserProgram.query.update(user_program, user_program_data)
```
Because if there was user_program then obviously there was user_program.program_id
 Can we go with this?

Reply:
If user_program.program_id cannot be NULL at this point then I think the second logic will work. Just confirm once if there is any case where `user_program.program_id` is NULL